### PR TITLE
feat(copilot): add integration

### PR DESCRIPTION
**Pull request by joshmedeski**
Adding [Github's oficial copilot.nvim repo](https://github.com/github/copilot.vim).

### CODE REVIEW

File: lua/catppuccin/types.lua [hunk 1]
```diff
@@ -105,6 +105,7 @@
 ---@class CtpIntegrations
 ---@field aerial boolean?
 ---@field alpha boolean?
+---@field copilot_vim boolean?
```
Comment (sgoudham) [inline lua/catppuccin/types.lua:108]: Please move this to the correct line considering it's in alphabetical order.

File: lua/catppuccin/groups/integrations/copilot_vim.lua [hunk 2]
```diff
@@ -0,0 +1,10 @@
+local M = {}
+
+function M.get()
+	return {
+		CopilotSuggestion = { fg = C.subtext0 },
+		CopilotAnnotation = { fg = C.blue },
+	}
+end
```
Comment (sgoudham) [inline lua/catppuccin/groups/integrations/copilot_vim.lua:8]: Could you please provide before/after screenshots of this change?

**Review by sgoudham — CHANGES_REQUESTED** [review]

File: lua/catppuccin/groups/integrations/copilot_vim.lua (hunk 2 above)
Comment (joshmedeski) [reply lua/catppuccin/groups/integrations/copilot_vim.lua:8]: Here is the after

SCR-20241113-oflv

I'm having a hard time getting the highlight group to reset, but I believe all text is set to a `#555555` color.

File: lua/catppuccin/groups/integrations/copilot_vim.lua [hunk 3]
```diff
@@ -0,0 +1,10 @@
+local M = {}
+
+function M.get()
+	return {
+		CopilotSuggestion = { fg = C.subtext0 },
```
Comment (sgoudham) [inline lua/catppuccin/groups/integrations/copilot_vim.lua:5]: I've had some discussion with some other maintainers and we'll want to change this / standardise on `overlay0` for AI/Copilot inline completions.

**Review by sgoudham — CHANGES_REQUESTED** [review]
Just one comment for the code and you'll also need to update the [integrations markdown table listed on the README](https://github.com/catppuccin/nvim#integrations) to reflect this update.

File: lua/catppuccin/groups/integrations/copilot_vim.lua (hunk 3 above)
Comment (joshmedeski) [reply lua/catppuccin/groups/integrations/copilot_vim.lua]: Good idea, I updated it.

File: lua/catppuccin/init.lua [hunk 4]
```diff
@@ -40,6 +40,7 @@ local M = {
 		integrations = {
 			alpha = true,
 			cmp = true,
+			copilot_vim = true,
```
Comment (sgoudham) [inline lua/catppuccin/init.lua:43]: Just one more change sorry, I don't believe we're adding more integrations defaulting to `true` anymore. I'd appreciate if you could remove this line.

File: README.md [hunk 5]
```diff
@@ -1047,6 +1047,20 @@ cmp = true
...
 <!-- nvim-cmp -->
 
+<!-- nvim-copilot-vim -->
+</tr>
+<tr>
+<td> <a href="https://github.com/github/copilot.vim">copilot.vim</a> </td>
+<td>
+
+```lua
+copilot_vim = true,
```
Comment (sgoudham) [inline README.md:1057]: Should be set to `false` since it shouldn't be enabled by default as explained below.

```suggestion
copilot_vim = false,
```

**Review by sgoudham — CHANGES_REQUESTED** [review]
One more comment, sorry!

File: lua/catppuccin/init.lua (hunk 4 above)
Comment (joshmedeski) [reply lua/catppuccin/init.lua]: No problem, totally understand.

**Review by vollowx — APPROVED** [review]
LGTM!

**Pull request merged**